.github/workflows/pytest.yml

```diff
@@ -31,16 +30,23 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v3
+      # We check out only a limited depth and then pull tags to save time
+      - name: Checkout source
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 100
+
+      - name: Get tags
+        run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
 
-      - name: Set up conda environment
+      - name: Install Conda environment with Micromamba
         if: matrix.environment-type == 'conda'
-        uses: conda-incubator/setup-miniconda@v2
+        uses: mamba-org/provision-with-micromamba@v12
         with:
-          activate-environment: test_env
           environment-file: ${{ env.REQUIREMENTS }}
-          python-version: ${{ matrix.python-version }}
-          auto-activate-base: false
+          cache-downloads: true
```

Comment:
A bit of interesting info here regarding caching in case anyone is interested: https://github.com/mamba-org/provision-with-micromamba/pull/38#discussion_r808837618

Reply:
Tell me if you want to enable cache-env, stay with cache-downloads or removing cache alltogether